mithril-stm/src/aggregate_signature/basic_verifier.rs

```diff
@@ -232,13 +239,15 @@ impl BasicVerifier {
             parameters,
             msg,
             &sig_reg_list,
-        )?;
+        )
+        .with_context(|| "Basic verification failed during selection of unique k indices.")?;
 
         Self::preliminary_verify(&self.total_stake, &unique_sigs, parameters, msg)?;
 
         let (sigs, vks) = Self::collect_signatures_verification_keys(&unique_sigs);
 
-        BlsSignature::verify_aggregate(msg.to_vec().as_slice(), &vks, &sigs)?;
+        BlsSignature::verify_aggregate(msg.to_vec().as_slice(), &vks, &sigs)
+            .with_context(|| "Basic verifier failed in multisignature verification.")?;
```

Comment:
Maybe this is a bit more clear:
```suggestion
            .with_context(|| "Basic verifier failed during BLS aggregate signature verification.")?;
```